src/routes/index.ts

```diff
@@ -286,24 +287,26 @@ router.route(config.UPDATE_OVERSEAS_ENTITY_CONFIRM_URL)
     navigation.hasOverseasEntity
   )
   .get(confirmOverseasEntityDetails.get)
-  .post(confirmOverseasEntityDetails.post);
+  .post(companyAuthentication, confirmOverseasEntityDetails.post);
 
 router.route(config.OVERSEAS_ENTITY_PRESENTER_URL)
   .all(
     authentication,
+    companyAuthentication,
```

Comment:
I do not know if we need to add `companyAuthentication` in each endpoint if we have a navigation check in place, probably check it the first time when the company is posted and use the navigation logic already in place

Reply:
@mouhajer-ch , had followed the approach of adding it only on the first endpoint but was suggested adding it to all as an extra security measure in case navigation checks are missed.